sql/parser/sql.y

```diff
@@ -1003,19 +1003,17 @@ set_reset_clause:
   SET set_rest {}
 
 show_stmt:
-  SHOW var_name
-  {
-    // It would be cleaner if we could have "SHOW DATABASES" and "SHOW
-    // TABLES" rules, but unfortunately DATABASES and TABLES are
-    // unreserved keywords and thus valid variable names and such
-    // rules would cause reduce/reduce conflicts.
-    if strings.EqualFold($2.String(), `"DATABASES"`) {
-      $$ = &ShowDatabases{}
-    } else if strings.EqualFold($2.String(), `"TABLES"`) {
-      $$ = &ShowTables{}
-    } else {
-      $$ = nil
```

Comment:
ah. so that's what it's for. I'd say let's worry about it when we get around to it.
However, var_name allows unreserved keywords. It seems iffy to let someone set a variable called `DATABASES` (for example). Using a slightly stricter set would allow us to keep the individual SHOW cases.